organization/ircam_oauth2_provider/views.py

```diff
@@ -0,0 +1,114 @@
+import requests
+from allauth.socialaccount.models import (SocialAccount,SocialLogin)
+from allauth.socialaccount.providers.oauth2.views import (OAuth2Adapter, OAuth2LoginView, OAuth2CallbackView)
+from .provider import IrcamAuthProvider
+from django.conf import settings
+from django.contrib.auth.models import User
+from organization.network.models import Person
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import IntegrityError
+
+import logging
+
+class IrcamAuthAdapter(OAuth2Adapter):
+
+    if not hasattr(settings, 'OAUTH_SERVER_BASEURL'):
+        raise Exception("Couldn't find OAUTH_SERVER_BASEURL in the settings")
+    
+    logger = logging.getLogger('ircamauth')
+    logger.addHandler(logging.StreamHandler())
+    logger.setLevel(logging.DEBUG) if settings.DEBUG else logger.setLevel(logging.INFO)
+   
+    provider_id = IrcamAuthProvider.id
+    access_token_url = '{}/o/token/'.format(settings.OAUTH_SERVER_BASEURL)  # Called programmatically, must be reachable from container
+    authorize_url = '{}/o/authorize/'.format(settings.USER_SERVER_BASEURL)  # Accessed by the client so must be host-reachable
+    profile_url = '{}/profile/'.format(settings.OAUTH_SERVER_BASEURL)
+
+    def create_or_update_person(self,user):
+        try :
+            person, created = Person.objects.get_or_create(user_id=user.id)            
+            person.user = user
+            person.title = user.first_name+" "+user.last_name
+            person.first_name = user.first_name
+            person.last_name = user.last_name
+            person.email = user.email
+
+            person.save()
+
+        except ObjectDoesNotExist:
+            person = Person.objects.get(
+                user_id=user.id
+            )
+        except IntegrityError:
+            pass
+
+    def create_user_socialaccount(self,request,extra_data):
+
+        user, created = User.objects.update_or_create(
+            username = extra_data['username'],
+            defaults={
+                'first_name': extra_data['first_name'],
+                'last_name': extra_data['last_name'],
+                'is_active': True,
+                'is_superuser': False,
+                'is_staff': False,
+                'email': extra_data['email'],
+            }
+        )
+        user.save()
+        self.logger.info('Local user {0} id:{1} username:{2} email:{3}'.format('created: ' if created else 'updated: ',user.id,user.username,user.email))
```

Comment:
It leaks user data, it should be a debug log.

Reply:
use `if settings.DEBUG:`

Reply:
I'd rather use `logger.debug()`.
When the app is configured for production it should discard logs with `debug` level.

Reply:
The idea was to trace user creation and connection in production. If we don't need it in production let me know.